packages/edit-site/src/components/block-list-exploded/index.js

```diff
@@ -1,14 +1,21 @@
 /**
  * WordPress dependencies
  */
-import { store as blockEditorStore } from '@wordpress/block-editor';
+import { store as blockEditorStore, Inserter } from '@wordpress/block-editor';
 import { useSelect } from '@wordpress/data';
+import { pure } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import BlockListExplodedItem from './item';
 
+const InserterBeforeBlock = pure( ( { clientId } ) => (
```

Comment:
I didn't know we had this HoC. Should I prefer using this over `React.memo`?

Reply:
To be honest at this point, I don't know. What's certain is that we need to make a decision :P. This HoC was here before React introduced React.memo and if I'm not wrong also works for Class component contrary to React.memo which I think is only suited for function components.

Reply:
Right, React.memo works only with functional components.

> What's certain is that we need to make a decision

Got it. Use `React.memo` with function components use `pure` with Class components 😄